### PR TITLE
HYDRA-712 : Handle hierarchical transform & visibility changes for plugin nodes

### PR DESCRIPTION
**Pull request by debloip-adsk**

### CODE REVIEW

File: test/lib/mayaUsd/render/mayaToHydra/FlowPluginsHierarchicalPropertiesTest/usdStageAnimatedPrim_t15.png
Comment (lanierd-adsk) [inline test/lib/mayaUsd/render/mayaToHydra/FlowPluginsHierarchicalPropertiesTest/usdStageAnimatedPrim_t15.png:1]: Is this image being empty expected ?

File: test/lib/mayaUsd/render/mayaToHydra/FlowViewportAPITest/hierarchicalProperties_locator_visibility_on.png
Comment (lanierd-adsk) [inline test/lib/mayaUsd/render/mayaToHydra/FlowViewportAPITest/hierarchicalProperties_locator_visibility_on.png:1]: Are the cubes being semi-transparent expected ?

File: test/lib/mayaUsd/render/mayaToHydra/FlowPluginsHierarchicalPropertiesTest/usdStageAnimatedPrim_t15.png
Comment (debloip-adsk) [reply test/lib/mayaUsd/render/mayaToHydra/FlowPluginsHierarchicalPropertiesTest/usdStageAnimatedPrim_t15.png:1]: Yep, I had added a comment to that effect here : https://github.com/Autodesk/maya-hydra/pull/141/files/c137ac27c3bf7a6941ca4626ca31453e75c8eced#diff-c4607392b118ce011b62fa5a94e6681c03986cfed357b9e5730e30a13b0e6ad8R188
You can also see the USD timeSamples here : https://github.com/Autodesk/maya-hydra/pull/141/files#diff-95f4f83179fe4db8124f53cc94c72138c1edccf2939dda366e81979f3918c8baR29-R32

File: test/lib/mayaUsd/render/mayaToHydra/FlowViewportAPITest/hierarchicalProperties_locator_visibility_on.png
Comment (debloip-adsk) [reply test/lib/mayaUsd/render/mayaToHydra/FlowViewportAPITest/hierarchicalProperties_locator_visibility_on.png:1]: Huh I hadn't even noticed they're transparent. But this seems expected, as the cubes' opacity is set to 0.8, see here : 
https://github.com/Autodesk/maya-hydra/blob/dev/lib/flowViewport/API/samples/fvpDataProducerSceneIndexExample.h#L76
and here : 
https://github.com/Autodesk/maya-hydra/blob/dev/lib/flowViewport/API/samples/fvpDataProducerSceneIndexExample.h#L105-L106
This is also visible in the scene index : 
image
And the reference snapshots for testFlowViewportAPI also show semi-transparent opacity.

However, in that case what I can't explain is why the cubes are _not_ transparent when in the actual viewport. So in that sense it seems that it is the viewport behaviour that is incorrect, and not the snapshots. I logged HYDRA-1097 to track that. Good observation!

**Review by lanierd-adsk — DISMISSED** [review]

**Review by lilike-adsk — APPROVED** [review]

**Pull request merged**